### PR TITLE
s390x: Enable "make deb" for Ubuntu Xenial (16.04)

### PR DESCRIPTION
**Pull request by michael-holzheu**
With this patch and Docker PR #25883 ("Add a Dockerfile for generating manpages on s390x") "make deb" creates the following packages for s390x:

``` bash
 # cd bundles/1.13.0-dev/build-deb/
 # find .
   .
   ./ubuntu-xenial
   ./ubuntu-xenial/docker-engine_1.13.0~dev~git20160823.161729.0.2693af4-0~xenial_s390x.deb
   ./ubuntu-xenial/docker-engine_1.13.0~dev~git20160823.161729.0.2693af4-0~xenial.dsc
   ./ubuntu-xenial/docker-engine_1.13.0~dev~git20160823.161729.0.2693af4-0~xenial_s390x.changes
   ./ubuntu-xenial/Dockerfile.build
   ./ubuntu-xenial/docker-engine_1.13.0~dev~git20160823.161729.0.2693af4-0~xenial.tar.gz
   ./docker.log
   ./test.log
```

Package "docker-engine_1.13.0~dev~git20160823.161729.0.2693af4-0~xenial_s390x.deb"
could be successfully installed on a s390x Ubuntu system:

``` bash
 # cat /etc/issue
 Ubuntu 16.04.1 LTS \n \l
 # dpkg -i docker-engine_1.13.0~dev~git20160823.161729.0.2693af4-0~xenial_s390x.deb
 ...
 Installing new version of config file /etc/init.d/docker ...
 Installing new version of config file /etc/init/docker.conf ...
 Processing triggers for systemd (229-4ubuntu7) ...
 Processing triggers for ureadahead (0.100.0-19) ...
 Processing triggers for man-db (2.7.5-1) ...

 # docker version
 Client:
  Version:      1.13.0-dev
  API version:  1.25
  Go version:   go1.7
  Git commit:   2693af4-unsupported
  Built:        Wed Aug 24 11:41:13 2016
  OS/Arch:      linux/s390x

 Server:
 ...
```

The s390x "generate.sh" is a modified version of "ppc64le/generate.sh". We removed seccomp for s390x because we need at least libseccomp version 2.3.1 which is not provided by Ubuntu Xenial.

Signed-off-by: Michael Holzheu <holzheu@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by michael-holzheu** [comment]
Ping @tophj-ibm since you did the ppc64le part: 64881dc331f1f0ff861eb82bb05eef63d6693a67

File: contrib/builder/deb/s390x/generate.sh [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	
+	# update and install packages
+	echo "RUN apt-get update && apt-get install -y ${packages[*]} --no-install-recommends && rm -rf /var/lib/apt/lists/*" >> "$version/Dockerfile"
+	echo >> "$version/Dockerfile"
+
+	# s390x doesn't have an official downloadable binary as of go 1.6.2. so use the
+	# older packaged go(v1.6.1) to bootstrap latest go, then remove the packaged go
+	awk '$1 == "ENV" && $2 == "GO_VERSION" { print; exit }' ../../../../Dockerfile.s390x >> "$version/Dockerfile"
+	echo 'ENV GO_DOWNLOAD_URL https://golang.org/dl/go${GO_VERSION}.src.tar.gz' >> "$version/Dockerfile"
+	echo 'ENV GOROOT_BOOTSTRAP /usr/lib/go-1.6' >> "$version/Dockerfile"
```
Comment (tophj-ibm) [inline contrib/builder/deb/s390x/generate.sh]: should this be 1.7 or is golang-go package on 16.04/z 1.6.x with all the z stuff backported?

Comment (michael-holzheu) [inline contrib/builder/deb/s390x/generate.sh]: > should this be 1.7 or is golang-go package on 16.04/z 1.6.x with all the z stuff backported?

Yes, the comment is wrong. What about:

```
# Use go version from Dockerfile.s390x
```

Comment (tophj-ibm) [inline contrib/builder/deb/s390x/generate.sh]: actually I was talking about /usr/lib/go-1.6 which comes from the golang-go package, I just wanted to make sure that was correct and it wasn't supposed to be 1.7 (although if you tested it and it worked, it should be)

**Comment by michael-holzheu** [comment]
I now updated the comment to:
```
# Get go version from Dockerfile.s390x and bootstrap with installed go version
```

**Comment by tophj-ibm** [comment]
seems good to me, ping @justincormack

**Comment by justincormack** [comment]
LGTM

**Comment by justincormack** [comment]
cc @tiborvass

**Comment by michael-holzheu** [comment]
ping @justincormack and @tiborvass : Any news?

**Comment by michael-holzheu** [comment]
FYI: I just submitted PR #26424. If this is accepted, I probably should also update this PR in order to use the binary go package.

**Comment by justincormack** [comment]
@michael-holzheu yes probably better to do that.

**Comment by michael-holzheu** [comment]
@justincormack I now updated the PR to  use the go binary package:

```bash
diff --git a/contrib/builder/deb/s390x/generate.sh b/contrib/builder/deb/s390x/generate.sh
index a063c38..a1f7729 100755
--- a/contrib/builder/deb/s390x/generate.sh
+++ b/contrib/builder/deb/s390x/generate.sh
@@ -76,10 +76,10 @@ for version in "${versions[@]}"; do
        echo "RUN apt-get update && apt-get install -y ${packages[*]} --no-install-recommends && rm -rf /var/lib/apt/lists/*" >> "$version/Dockerfile"
        echo >> "$version/Dockerfile"
 
-       # Get go version from Dockerfile.s390x and bootstrap with installed go version
-       awk '$1 == "ENV" && $2 == "GO_VERSION" { print; exit }' ../../../../Dockerfile.s390x >> "$version/Dockerfile"
-       echo 'ENV GO_DOWNLOAD_URL https://golang.org/dl/go${GO_VERSION}.src.tar.gz' >> "$version/Dockerfile"
-       echo 'ENV GOROOT_BOOTSTRAP /usr/lib/go-1.6' >> "$version/Dockerfile"
+       awk '$1 == "ENV" && $2 == "GO_VERSION" { print; exit }' ../../../../Dockerfile >> "$version/Dockerfile"
+       echo 'RUN curl -fSL "https://storage.googleapis.com/golang/go${GO_VERSION}.linux-amd64.tar.gz" | tar xzC /usr/local' >> "$version/Dockerfile"
+       echo 'ENV PATH $PATH:/usr/local/go/bin' >> "$version/Dockerfile"
+
        echo >> "$version/Dockerfile"
        
        echo 'RUN curl -fsSL "$GO_DOWNLOAD_URL" -o golang.tar.gz \' >> "$version/Dockerfile"
```

IMHO we can merge the PR now.

**Comment by michael-holzheu** [comment]
Hmm, sorry. I just realized that I forgot to rebuild `contrib/builder/deb/s390x/ubuntu-xenial/Dockerfile`...

**Comment by michael-holzheu** [comment]
> Hmm, sorry. I just realized that I forgot to rebuild contrib/builder/deb/s390x/ubuntu-xenial/Dockerfile...

Ok, I updated the PR. I hope now everything is good.

**Comment by michael-holzheu** [comment]
Ping @justincormack and @tiborvass : Ok for you?

**Comment by thaJeztah** [comment]
ping @justincormack PTAL

**Comment by justincormack** [comment]
LGTM

**Comment by thaJeztah** [comment]
LGTM

**Pull request merged**